src/components/paniniCreator/formSections/formSection/elements/elementsTypes.d.ts

```diff
@@ -1,13 +1,25 @@
 type SelectElementProps = {
   name: string;
   options: readonly string[];
-  formElementsValues: string[];
-  setFormElementsValues: React.Dispatch<React.SetStateAction<string[]>>;
-  orderVal: number;
-  defaultVal?: string;
-};
-
-interface SwipeElementProps extends SelectElementProps {
-  formElementsValues?: never;
-  children?: ReactNode;
-}
+  val: Record<"id", string>;
+} & (
+  | {
+      index: number;
+      update: UseFieldArrayUpdate<FieldValues, string>;
```

Comment:
**nitpick:** Yea, it got messy, and complicated. I think it's because you are trying to make it too generic. Don't hesitate to make more interfaces with more specific naming - it will be easier to work with. Such union types can lead to TS problems that are hard to solve. I would split it